design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,1023 @@
+    * [apic](http://kubevirt.io/api-reference/main/definitions.html#_v1_featureapic)
+      * `VirtualMachinePreferenceSpec.Features.PreferredApic`
+    * [hyperv](http://kubevirt.io/api-reference/main/definitions.html#_v1_featurehyperv) 
+      * `VirtualMachinePreferenceSpec.Features.PreferredHyperv`
+    * [kvm](http://kubevirt.io/api-reference/main/definitions.html#_v1_featurekvm)
+      * `VirtualMachinePreferenceSpec.Features.PreferredKvm`
+    * [pvspinlock](http://kubevirt.io/api-reference/main/definitions.html#_v1_featurestate) 
+      * `VirtualMachinePreferenceSpec.Features.PreferredPvspinlock`
+    * [smm](http://kubevirt.io/api-reference/main/definitions.html#_v1_featurestate)
+      * `VirtualMachinePreferenceSpec.Features.PreferredSmm`
+  * [Firmware](http://kubevirt.io/api-reference/main/definitions.html#_v1_firmware)
+    * [bootloader](http://kubevirt.io/api-reference/main/definitions.html#_v1_bootloader)
+      * [bios](http://kubevirt.io/api-reference/main/definitions.html#_v1_bios)
+        * `VirtualMachinePreferenceSpec.Firmware.PreferredUseBios`
+        * useSerial
+          * `VirtualMachinePreferenceSpec.Firmware.PreferredUseBiosSerial`
+      * [efi](http://kubevirt.io/api-reference/main/definitions.html#_v1_efi)
+        * `VirtualMachinePreferenceSpec.Firmware.PreferredUseEfi`
+        * secureBoot
+          * `VirtualMachinePreferenceSpec.Firmware.PreferredUseEfiSecureBoot`
+    * [kernelBoot](http://kubevirt.io/api-reference/main/definitions.html#_v1_kernelboot)
+      * [container](http://kubevirt.io/api-reference/main/definitions.html#_v1_kernelbootcontainer)
+        * image
+          * None
+        * imagePullPolicy
+          * None
+        * imagePullSecret
+          * None
+        * initrdPath
+          * None
+        * kernelPath
+          * None
+      * kernelArgs
+        * None
+    * serial
+      * None
+    * uuid
+      * None
+  * ioThreadsPolicy
+    * `VirtualMachineFlavorSpec.ioThreadsPolicy`
+  * [launchSecurity](http://kubevirt.io/api-reference/main/definitions.html#_v1_launchsecurity)
+    * [sev](http://kubevirt.io/api-reference/main/definitions.html#_v1_sev)
+      * `VirtualMachineFlavorSpec.LaunchSecurity.SEV`
+  * [Machine](http://kubevirt.io/api-reference/main/definitions.html#_v1_machine)
+    * type
+      * `VirtualMachinePreferenceSpec.Machine.PreferredMachineType`
+  * [Memory](http://kubevirt.io/api-reference/main/definitions.html#_v1_memory)
+    * [guest](http://kubevirt.io/api-reference/main/definitions.html#_k8s_io_apimachinery_pkg_api_resource_quantity) 
+      * `VirtualMachineFlavorSpec.Memory.Guest`
+    * [hugepages](http://kubevirt.io/api-reference/main/definitions.html#_v1_hugepages)
+      * `VirtualMachineFlavorSpec.Memory.HugePages`
+
+## Types
+
+### `VirtualMachineFlavorSpec`
+
+```go
+// VirtualMachineFlavorSpec
+//
+// +k8s:openapi-gen=true1
+type VirtualMachineFlavorSpec struct {
+	CPU *CPUFlavor `json:"cpu"`
+
+	Memory *MemoryFlavor `json:"memory"`
```

Comment:
Nit pick, and not relevant in the design, but for API documentation this field should get some comment as well.

Reply:
ACK I'll add these in the series I have posted and update the design proposal.